Sources/SolanaSwift/BlockchainClient/Helpers/Borsh/BinaryReader.swift

```diff
@@ -61,16 +62,37 @@ public extension BinaryReader {
     }
 
     mutating func decodeLength() throws -> Int {
-        var len: UInt8 = 0
-        var size: UInt8 = 0
+        var len: UInt = 0
+        var shift: UInt = 0
+        let maxBytes = 10 // Maximum bytes for 64-bit LEB128
+
         while true {
+            guard shift < (maxBytes * 7) else {
+                throw BinaryReaderError.invalidDecodeLength
+            }
+
             let elem: UInt8 = try read()
-            len |= (elem & 0x7F) << (size * 7)
```

Comment:
Shifting a UInt8 beyond 7 bits doesn't accumulate the intended bits as expected.